artifactory/system.go

```diff
@@ -33,6 +40,1112 @@ func (v Versions) String() string {
+	ArtifactoryPort          *int    `yaml:"artifactoryPort,omitempty" xml:"artifactoryPort,omitempty"`
+}
+
+// ReverseProxies is an alias for a slice of ReverseProxy that can be
+// properly marshaled to/from YAML.
+type ReverseProxies []*ReverseProxy
+
+// MarshalYAML implements the yaml.Marshaller interface for ReverseProxies.
+func (r ReverseProxies) MarshalYAML() (interface{}, error) {
+	reverseProxiesMap := make(map[string]*ReverseProxy)
+	for _, reverseProxy := range r {
+		if *reverseProxy == (ReverseProxy{Key: reverseProxy.Key}) {
+			reverseProxiesMap[*reverseProxy.Key] = nil
+		} else {
+			reverseProxiesMap[*reverseProxy.Key] = reverseProxy
+		}
+	}
+
+	return reverseProxiesMap, nil
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler for ReverseProxies.
+func (r *ReverseProxies) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	reverseProxiesMap := make(map[string]*ReverseProxy)
+	if err := unmarshal(reverseProxiesMap); err != nil {
+		return err
+	}
+
+	var reverseProxiesSlice ReverseProxies
+	for reverseProxyKey, reverseProxy := range reverseProxiesMap {
+		if reverseProxy == nil {
+			continue
+		}
+		reverseProxy.Key = &reverseProxyKey
+		reverseProxiesSlice = append(reverseProxiesSlice, reverseProxy)
+	}
+
+	*r = reverseProxiesSlice
+	return nil
+}
+
+// PropertySet represents a Property Set in Artifactory's Global Configuration Descriptor.
+//
+// Docs: https://www.jfrog.com/confluence/display/RTF/YAML+Configuration+File#YAMLConfigurationFile-General(General,PropertySets,Proxy,Mail)
+type PropertySet struct {
+	Name       *string     `yaml:"-" xml:"name,omitempty"`
+	Properties *Properties `yaml:"properties,omitempty" xml:"properties>property,omitempty"`
+	Visible    *bool       `yaml:"visible,omitempty" xml:"visible,omitempty"`
+}
+
+// Properties is an alias for a slice of Property in a PropertySet that can be
+// properly marshaled to/from YAML.
+type Properties []*Property
+
+// MarshalYAML implements the yaml.Marshaller interface for Properties.
+func (p Properties) MarshalYAML() (interface{}, error) {
+	propertiesMap := make(map[string]*Property)
+	for _, property := range p {
+		if *property == (Property{Name: property.Name}) {
+			propertiesMap[*property.Name] = nil
+		} else {
+			propertiesMap[*property.Name] = property
+		}
+	}
+
+	return propertiesMap, nil
+}
+
+// UnmarshalYAML implements yaml.Unmarshaler for Properties.
+func (p *Properties) UnmarshalYAML(unmarshal func(interface{}) error) error {
+	propertiesMap := make(map[string]*Property)
+	if err := unmarshal(propertiesMap); err != nil {
+		return err
+	}
+
+	var propertiesSlice Properties
+	for propertyName, property := range propertiesMap {
+		if property == nil {
+			continue
+		}
+		property.Name = &propertyName
+		propertiesSlice = append(propertiesSlice, property)
+	}
+
+	*p = propertiesSlice
+	return nil
+}
+
+type Property struct {
+	Name                   *string           `yaml:"-" xml:"name,omitempty"`
```

Comment:
I noticed we're not allowing this field to be serialized in YAML due to the `-` tag.

This peaked my curiosity so was this intentional?